### ISSUE TITLE
Make re-entrant

### ISSUE THREAD

**Issue by rpherbig** [rpherbig/auctioneer#1]
Save IDs to text file and read on run?

**Comment by jackson-byrd** [comment]
I'm going to create this method for the subscriber list, so I'll take this on.

**Comment by jackson-byrd** [comment]
Actually we could knock out this, #12 and #13 if we just create a main JSON with everyone who subscribes or adds their leadership and essence.

**Comment by rpherbig** [comment]
I think #12 and #13 need a bit more work than _just_ this, ideally they'd have get/set commands so you/I don't have to manually update things.

My thought was to keep this separate from leadership/essence, and once it's done make #19 the next priority. That minimizes the need for restarting the bot for any _data_ reasons; and makes it easier to restart after _code_ changes.

**Comment by jackson-byrd** [comment]
Good point on extra work needed, I was thinking that we could have like a JSON file organized like:
`{
    "discordName": "foo",
    "discordID" : 12345909808098,
    "subscribed": true,
    "speciesWar": {
        "currentEssences": 0,
        "currentLeadership": 130
    }
    "auction": {
        "fieldName": "",
    },
    "someOtherBotFeature": {
    
    }
}`

So we don't create a new file for every feature

**Comment by rpherbig** [comment]
Agreed re: one file. However, I think at the top-level we're going to have two buckets: channel info and user info. Channel info will be things like "for channel X, monitor messages A, B, C". At least, that's how the _auction_ info is organized today (basically, everything that is in an instance var (`@foo`) will be in the JSON file).

**Comment by jackson-byrd** [comment]
This is kind of what I have so far as far as the layout of the JSON file, not sure exactly what you wanted for ChannelInfo but threw something together.  The big problem right now is trying to access based on `discordId` and `channelId` as Ruby complains about numbers that big.  I'm likely going to have to implement our own ID system, unless you have a better idea?

@rpherbig 

`{
    "userInfo": 
        {
            "discordId": 199227044821860352,
            "discordName": "onyxarm",
            "rentInfo": true
        },
    "channelInfo": 
        {
            "channelId": 1157369992245952532,
            "channelName": "war-auction",
            "channelType": "text"
        }
}`

**Comment by rpherbig** [comment]
Huh, I wouldn't expect Ruby to complain about that since they're Integers internally. Maybe store the IDs as strings and convert to Integer after parsing? Can you post the specific error here?

As for the structure, I wouldn't worry about channel info for now. Let's get the file work in place and then we can retrofit it.

**Comment by rpherbig** [comment]
Ok, I know we talked about the file being split by users/channels, but I've been pondering it this morning and I think the top-level being per-feature would be the easiest for us to maintain long-term.

```JSON
{
    auctions: [{
        channel: 1234,
        messages: [...]
    }, {...}],
    featureTwo: {...}
}
```

**Comment by jackson-byrd** [comment]
I really like this idea.  I'll get to work on this and let you know when I got something workable

**Comment by jackson-byrd** [comment]
I kinda get how most of the auction bot works, but what messages do you want to put in the data file?

Also going to my earlier problem, I was trying to do something incorrectly, I don't think it's a problem.  I blame coding while drinking lol.

**Comment by rpherbig** [comment]
My goal is for all state to be written to the file, so at any point I can kill & rerun the bot and have it pick back up where it was
